### ISSUE TITLE
project bar, folder collapse search issue

### ISSUE THREAD

**Issue by F3der1co** [Scirra/Construct-bugs#7047]
## Problem description
when searching for a folder int he project bar its contents don't show up, which is fine, but when pressing Escape to see the content the view jumps back to the top and subfolders stay collapsed, making looking for folders via the search borderline impossible, as you are again have to search for it by hand.

https://github.com/Scirra/Construct-bugs/assets/89512047/3ec298da-c8ec-4076-be9e-17cbf7168bb8

## Attach a .c3p
[folderSearchIssue.zip](https://github.com/Scirra/Construct-bugs/files/11489733/folderSearchIssue.zip)

## Steps to reproduce

1. search for 2.1.1
2. select the folder
3. press escape to be able to see the content of the folder

## Observed result

Escape resets to pre-search collapse state and jumps back to the top

## Expected result

the selected folder/object is scrolled to and if inside folders/sub-folders they get uncollapsed

## More details
Because of the current design you will basically have to search through the project by hand again and the search is not of use at all in this case. In my larger projects I often search for folders, for example I want to place some foliage, I search for the foliage folder and want to see all the assets in it to pick from them, so searching for a specific object type is not what I want.

This is a somewhat similar issue https://github.com/Scirra/Construct-bugs/issues/7021, which just got fixed, but in a different context.

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by AshleyScirra** [comment]
I'd characterise this as a suggestion rather than a bug, especially as this is not necessarily an obvious change: search should not be destructive for the current expand/collapse folder state. For example if you search for just `e` which matches most things in the project and expands most folders, then clear the search, you don't want to be left with most of the Project Bar left expanded. So I don't think this is a case of an obviously-wrong thing with an obviously-correct solution. I also want to discourage posting suggestions to the issue tracker (especially as some people take that as a sign they should post suggestions to the issue tracker anyway rather than the suggestions platform, which is not what we want), so please file this as a suggestion instead.

**Comment by F3der1co** [comment]
Just be be clear why I filed it as a bug instead of a suggestion: 
With the current implementation it is possible to search for folders, but the search results are not usable in any meaningful way. So to me the this is a feature that exists but in a broken/unusable way.

Also what I was meaning was not to keep every folder expanded that was part of the search, but instead expanding the folders corresponding to the selection, so the selected object/folder remains focused and visible! (maybe ESCAPE could keep the current behavior and ENTER does this)

I will file a suggestion whenever I get around to it.

**Comment by F3der1co** [comment]
I see, yea than the clicked one makes more sense.

**Comment by OverboyDev** [comment]
Agreed. On a similar topic, it would be great if we could just expand explicitely searched Folders and Families directly within the Search Mode as this is the most common situation where it's frustrated that the Searched Item isn't restored when leaving the search mode.

**Ability to expand searched Subfolders/Families :** https://construct23.ideas.aha.io/ideas/C23-I-300

Both suggestions can solve similar issue but the expected behavior is that both of them work : 
- Ability to expand searched stuff (Folders and Family) within the Search Mode
- Keep the searched item even while leaving the Search mode.
